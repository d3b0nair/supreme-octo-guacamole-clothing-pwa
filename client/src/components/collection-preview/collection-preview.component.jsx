import React, { useState } from "react";
import { withRouter } from "react-router-dom";

import CollectionItem from "../collection-item/collection-item.component";
import CustomButton from "../custom-button/custom-button.component";
import {
  TitleContainer,
  PreviewContainer,
  CollectionPreviewContainer,
} from "./collection-preview.styles";

export function CollectionPreview({ title, items, history, match, routeName }) {
  const [itemsToLoad, setItemsToLoad] = useState(4);
  const [showingMore, setShowingMore] = useState(false);
  const showLess = () => {
    setItemsToLoad(5);
    setShowingMore(!showingMore);
  };
  const showMore = () => {
    setItemsToLoad(items.length);
    setShowingMore(!showingMore);
  };
  return (
    <CollectionPreviewContainer>
      <TitleContainer
        data-testid="TitleContainer"
        onClick={() => history.push(`${match.path}/${routeName}`)}
      >
        {title.toUpperCase()}
      </TitleContainer>
      <PreviewContainer>
        {items
          .filter((item, idx) => idx < itemsToLoad)
          .map((item) => (
            <CollectionItem key={item.id} item={item} />
          ))}
      </PreviewContainer>
      <CustomButton
        inverted
        style={{ margin: "3vh 0 3vh 0" }}
        onClick={() => {
          showingMore ? showLess() : showMore();
        }}
      >
        {showingMore ? "SHOW LESS" : `SHOW MORE ${title.toUpperCase()}`}
      </CustomButton>
    </CollectionPreviewContainer>
  );
}

export default withRouter(CollectionPreview);
