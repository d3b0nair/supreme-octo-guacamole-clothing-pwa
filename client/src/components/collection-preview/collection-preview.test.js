import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from "@testing-library/react";
import { CollectionPreview } from "./collection-preview.component";

const mockPush = jest.fn();
const mockProps = (otherProps) => ({
  ...otherProps,
  title: "",
  items: [],
  history: { push: mockPush },
  match: { path: jest.fn() },
  routeName: "",
});
const component = (mockProps) => <CollectionPreview {...mockProps} />;

afterEach(cleanup);

describe("CollectionItem component", () => {
  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
  it("should call addItem on button click", async () => {
    render(component(mockProps()));
    await waitFor(() => fireEvent.click(screen.getByTestId("TitleContainer")));
    expect(mockPush).toHaveBeenCalled();
  });
});
