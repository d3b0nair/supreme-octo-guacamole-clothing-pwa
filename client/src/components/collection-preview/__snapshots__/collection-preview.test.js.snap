// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should mount 1`] = `
<div>
  <div
    class="sc-iCfMLu cChwTn"
  >
    <h1
      class="sc-furwcr jHEmAL"
      data-testid="TitleContainer"
    />
    <div
      class="sc-pVTFL ifDaVZ"
    />
    <button
      class="sc-bdvvtL gRKetO"
      style="margin: 3vh 0px 3vh 0px;"
    >
      SHOW MORE 
    </button>
  </div>
</div>
`;
